src/rust/engine/process_execution/src/lib.rs

```diff
@@ -470,6 +484,14 @@ impl From<ProcessResultMetadata> for ExecutedActionMetadata {
   }
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ProcessResultSource {
+  RanLocally,
+  RanRemotely,
+  HitLocally,
+  HitRemotely,
```

Comment:
Naming: Might be better named as `CachedLocally` and `CachedRemotely`. "hit" is a bit ambiguous without reference to caching.

Reply:
`Cached` is potentially ambiguous with a write to the cache... will stick to this terminology.